File: backend/app/models/game.py
from pydantic import BaseModel
from typing import List
from enum import Enum

class DifficultyLevel(str, Enum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"

class Level(BaseModel):
    level_id: int
    difficulty: DifficultyLevel
    bottles: List[List[str]]
    num_colors: int
    num_bottles: int
    max_moves: int

File: frontend/src/types/game.ts
export interface Level {
  level_id: number;
  difficulty: 'easy' | 'medium' | 'hard';
  bottles: string[][];
  num_colors: number;
  num_bottles: number;
  max_moves: number;
}

export interface GameState {
  currentLevel: number;
  bottles: string[][];
  moves: number;
  maxMoves: number;
  isComplete: boolean;
}
